CHANGELOG.md

```diff
@@ -1,12 +1,16 @@
 # NEXT RELEASE
 
 ### Enhancements
-* <New feature description> (PR [#????](https://github.com/realm/realm-core/pull/????))
-* None.
+* Performance improvement for the following queries ([6376](https://github.com/realm/realm-core/issues/6376)):
+    * Reduced constant factor and memory use in Query::count() on simple equality queries for string/int/UUID/ObjectID when using an index. This is due to using `StringIndex::find_all_no_copy` instead of copying out all the result object keys into a buffer.
+    * Significant improvement on Timestamp equality queries when using an index.
```

Comment:
Could we add some numbers here? (and other places) It's not clear what Moderate and Significant really means?

Reply:
The numbers depend a lot on the data that is being queried, the query itself, and the number of results, but I added the percentage speedup I observed in our micro benchmarks.